src/compose/app.ts

```diff
@@ -657,33 +651,22 @@ export class App {
 			return false;
 		}
 
+		// Wait for networks to be created before starting the service
 		if (
 			_.some(
 				networkPairs,
-				(pair) =>
-					`${this.appId}_${pair.target?.name}` === target.config.networkMode,
+				(pair) => pair.target && target.hasNetworkMode(pair.target.name),
 			)
```

Comment:
Could just substitute for native method here as `networkPairs` is of type Array.
```
networkPairs.some((pair) => pair.target && target.hasNetworkMode(pair.target.name))
```

Reply:
yeah, that's a good point, I was trying to minimize the changes on the v3 PR but it makes sense to do it here